### PR TITLE
Lenient repository/homepage URL switch for packages that have no repository verification enabled.

### PR DESCRIPTION
**Pull request by isoos**
joFixes #1159.

### CODE REVIEW

File: lib/src/references/pubspec_urls.dart [hunk 1]
```diff
@@ -58,12 +59,24 @@ Future<PubspecUrlsWithIssues> checkPubspecUrls(PackageContext context) async {
...
-      pubspec.repository == null &&
-      isVerifiedRepository) {
-    var r = repository;
-    repository = homepage;
-    homepage = r;
+  if (pubspec.homepage != null && pubspec.repository == null) {
+    // We may switch these values if the repository has been verified, or
+    // if the verification was not enabled, but the URL parsing recognizes
+    // it as a valid repository.
+    var maySwitch = isVerifiedRepository;
```
Comment (sigurdm) [inline lib/src/references/pubspec_urls.dart:66]: Is switching not the wrong term here? Could we call this something like
```suggestion
    var canUseHomepageAsRepository = isVerifiedRepository;
```

**Review by sigurdm — APPROVED** [review]
LGTM

should we have a test covering this logic?

**Comment by isoos** [comment]
> should we have a test covering this logic?

In the long term this will be covered with the repository verification enabled for all packages. I can add a test or two, but if this breaks, no big deal for now.

**Comment by sigurdm** [comment]
> In the long term this will be covered with the repository verification enabled for all packages. I can add a test or two, but if this breaks, no big deal for now.

Yeah I guess it is not worth the time now - just was surprised that this broke in the first place.

Let's just ship it.

**Review by jonasfj — APPROVED** [review]

**Pull request merged**